File: OfflineFetcher/app/src/main/java/de/jbamberger/offlinefetcher/ui/reddit/RedditModule.java
package de.jbamberger.offlinefetcher.ui.reddit;


import android.app.Activity;

import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@Module//(subcomponents = RedditSubcomponent.class)
public abstract class RedditModule {

    @Binds
    @IntoMap
    @ActivityKey(RedditActivity.class)
    abstract AndroidInjector.Factory<? extends Activity>
    bindRedditActivityInjectorFactory(RedditSubcomponent.Builder builder);
}

File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/ui/reddit/RedditModule.kt
package de.jbamberger.offlineclient.ui.reddit


import dagger.Module
import dagger.android.ContributesAndroidInjector
import de.jbamberger.offlineclient.ui.reddit.stream.RedditStreamFragment
import de.jbamberger.offlineclient.ui.reddit.stream.RedditStreamScope

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@Module
abstract class RedditModule {

    @RedditStreamScope
    @ContributesAndroidInjector
    internal abstract fun contributeRedditStreamFragment(): RedditStreamFragment
}
